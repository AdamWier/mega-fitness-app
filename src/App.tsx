import * as React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import { ThemeProvider } from 'react-native-elements';
import {theme, navTheme} from './StyleSheet';
import Search from './screens/Search';
import Details from './screens/Details';
import Day from './screens/Day';

const Stack = createStackNavigator();

function App() {
  const [meal, updateMeal] = React.useState([]);

  return (
    <ThemeProvider theme={theme}>
      <NavigationContainer theme={navTheme}>
        <Stack.Navigator screenOptions={{
          headerStyle: {
            backgroundColor: '#375a7f',
          },
          headerTitleStyle: {
            color: '#fff',
          },
          headerTintColor: 'rgba(255, 255, 255, 0.6)',
        }}>
          <Stack.Screen name='Day'>
            {props => <Day {...props} meal={meal} />}
          </Stack.Screen>
          <Stack.Screen name='Search' component={Search} />
          <Stack.Screen name='Details' 
            options={({ route }) => ({ 
              title: route.params.details.name, 
            })}>
            {props => <Details {...props} meal={meal} updateMeal={updateMeal} />}
          </Stack.Screen>
        </Stack.Navigator>
      </NavigationContainer>
    </ThemeProvider>
  );
}

export default App;