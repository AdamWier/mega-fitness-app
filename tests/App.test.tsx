import React from 'react';
import renderer from 'react-test-renderer';
import App from '../src/App';

jest.mock('react-native-simple-toast', () => ({
  SHORT: 'SHORT',
}));

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

describe('<App />', () => {
  it('has 1 child', () => {
    expect.assertions(1);
    const tree = renderer
      .create(<App />)
      .toJSON() as renderer.ReactTestRendererJSON;
    expect(tree.children?.length).toBe(1);
  });

  it('renders correctly', () => {
    expect.assertions(1);
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
