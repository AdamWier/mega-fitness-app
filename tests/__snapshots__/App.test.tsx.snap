// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "zIndex": 1,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#375a7f",
                "borderBottomColor": "rgb(216, 216, 216)",
                "flex": 1,
                "shadowColor": "rgb(216, 216, 216)",
                "shadowOffset": Object {
                  "height": 0.5,
                  "width": 0,
                },
                "shadowOpacity": 0.85,
                "shadowRadius": 0,
              }
            }
          />
        </View>
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 64,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 20,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "marginHorizontal": 16,
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                aria-level="1"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "#ffffff",
                    "fontSize": 17,
                    "fontWeight": "600",
                  }
                }
              >
                Login
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <RNSScreenContainer
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        activityState={2}
        forwardedRef={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransition={[Function]}
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "overflow": undefined,
              },
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
            ]
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            needsOffscreenAlphaCompositing={false}
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "#222222",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "#222222",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "flex": 1,
                          "flexDirection": "column",
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "color": "#ffffff",
                            "fontSize": 50,
                            "textAlign": "center",
                          }
                        }
                        theme={
                          Object {
                            "Button": Object {
                              "buttonStyle": Object {
                                "backgroundColor": "#375a7f",
                                "borderRadius": 10,
                              },
                              "containerStyle": Object {
                                "backgroundColor": "#222222",
                                "margin": 10,
                              },
                            },
                            "Card": Object {
                              "containerStyle": Object {
                                "backgroundColor": "#222222",
                              },
                              "titleStyle": Object {
                                "color": "#ffffff",
                              },
                            },
                            "Header": Object {
                              "containerStyle": Object {
                                "borderBottomColor": "transparent",
                                "height": 56,
                                "paddingHorizontal": 0,
                                "paddingTop": 0,
                              },
                            },
                            "Icon": Object {
                              "color": "#ffffff",
                            },
                            "Input": Object {
                              "containerStyle": Object {
                                "backgroundColor": "#ffffff",
                                "marginHorizontal": 20,
                                "marginVertical": 10,
                                "padding": 0,
                                "width": 300,
                              },
                            },
                            "ListItem": Object {
                              "bottomDivider": true,
                              "chevron": true,
                              "containerStyle": Object {
                                "backgroundColor": "#222222",
                              },
                              "subtitleStyle": Object {
                                "color": "#ffffff",
                                "textAlign": "left",
                              },
                              "titleStyle": Object {
                                "textAlign": "left",
                              },
                            },
                            "Overlay": Object {
                              "borderRadius": 10,
                              "overlayStyle": Object {
                                "backgroundColor": "#222222",
                              },
                            },
                            "SearchBar": Object {
                              "containerStyle": Object {
                                "backgroundColor": "#222222",
                              },
                              "inputContainerStyle": Object {
                                "borderRadius": 0,
                              },
                              "inputStyle": Object {
                                "color": "#ffffff",
                              },
                            },
                            "Slider": Object {
                              "thumbTintColor": "#00bc8c",
                            },
                            "Text": Object {
                              "style": Object {
                                "color": "#ffffff",
                                "textAlign": "center",
                              },
                            },
                            "Tooltip": Object {
                              "withOverlay": false,
                            },
                            "colors": Object {
                              "background": "#222222",
                              "border": "rgb(216, 216, 216)",
                              "card": "rgb(255, 255, 255)",
                              "danger": "#E74C3C",
                              "disabled": "hsl(208, 8%, 90%)",
                              "divider": "#bcbbc1",
                              "error": "#ff190c",
                              "grey0": "#adb5bd",
                              "grey1": "#43484d",
                              "grey2": "#5e6977",
                              "grey3": "#86939e",
                              "grey4": "#bdc6cf",
                              "grey5": "#e1e8ee",
                              "greyOutline": "#bbb",
                              "info": "#3498DB",
                              "notification": "rgb(255, 59, 48)",
                              "platform": Object {
                                "android": Object {
                                  "error": "#f44336",
                                  "primary": "#2196f3",
                                  "secondary": "#9C27B0",
                                  "success": "#4caf50",
                                  "warning": "#ffeb3b",
                                },
                                "ios": Object {
                                  "error": "#ff3b30",
                                  "primary": "#007aff",
                                  "secondary": "#5856d6",
                                  "success": "#4cd964",
                                  "warning": "#ffcc00",
                                },
                              },
                              "primary": "#375a7f",
                              "searchBg": "#303337",
                              "secondary": "#8F0CE8",
                              "success": "#00bc8c",
                              "text": "#ffffff",
                              "warning": "#F39C12",
                            },
                          }
                        }
                      >
                        Login
                      </Text>
                      <View
                        style={
                          Object {
                            "backgroundColor": "#ffffff",
                            "marginHorizontal": 20,
                            "marginVertical": 10,
                            "padding": 0,
                            "paddingHorizontal": 10,
                            "width": 300,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "borderBottomWidth": 1,
                              "borderColor": "#86939e",
                              "flexDirection": "row",
                              "transform": Array [
                                Object {
                                  "translateX": 0,
                                },
                              ],
                            }
                          }
                        >
                          <TextInput
                            allowFontScaling={true}
                            editable={true}
                            onChangeText={[Function]}
                            placeholder="Email"
                            rejectResponderTermination={true}
                            style={
                              Object {
                                "alignSelf": "center",
                                "color": "black",
                                "flex": 1,
                                "fontSize": 18,
                                "minHeight": 40,
                              }
                            }
                            testID="RNE__Input__text-input"
                            underlineColorAndroid="transparent"
                          />
                        </View>
                      </View>
                      <View
                        style={
                          Object {
                            "backgroundColor": "#ffffff",
                            "marginHorizontal": 20,
                            "marginVertical": 10,
                            "padding": 0,
                            "paddingHorizontal": 10,
                            "width": 300,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "borderBottomWidth": 1,
                              "borderColor": "#86939e",
                              "flexDirection": "row",
                              "transform": Array [
                                Object {
                                  "translateX": 0,
                                },
                              ],
                            }
                          }
                        >
                          <TextInput
                            allowFontScaling={true}
                            editable={true}
                            onChangeText={[Function]}
                            placeholder="Password"
                            rejectResponderTermination={true}
                            secureTextEntry={true}
                            style={
                              Object {
                                "alignSelf": "center",
                                "color": "black",
                                "flex": 1,
                                "fontSize": 18,
                                "minHeight": 40,
                              }
                            }
                            testID="RNE__Input__text-input"
                            underlineColorAndroid="transparent"
                          />
                        </View>
                      </View>
                      <View
                        style={
                          Object {
                            "backgroundColor": "#222222",
                            "borderRadius": 10,
                            "margin": 10,
                          }
                        }
                      >
                        <View
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "opacity": 1,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "#375a7f",
                                "borderColor": "#375a7f",
                                "borderRadius": 10,
                                "borderWidth": 0,
                                "flexDirection": "row",
                                "justifyContent": "center",
                                "padding": 8,
                              }
                            }
                          >
                            <Text
                              style={
                                Object {
                                  "color": "white",
                                  "fontSize": 18,
                                  "paddingBottom": 1,
                                  "paddingTop": 2,
                                  "textAlign": "center",
                                }
                              }
                            >
                              Login
                            </Text>
                          </View>
                        </View>
                      </View>
                      <View
                        style={
                          Object {
                            "backgroundColor": "#222222",
                            "borderRadius": 10,
                            "margin": 25,
                          }
                        }
                      >
                        <View
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "opacity": 1,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "#3498DB",
                                "borderColor": "#375a7f",
                                "borderRadius": 10,
                                "borderWidth": 0,
                                "flexDirection": "row",
                                "justifyContent": "center",
                                "padding": 8,
                              }
                            }
                          >
                            <Text
                              style={
                                Object {
                                  "color": "white",
                                  "fontSize": 18,
                                  "paddingBottom": 1,
                                  "paddingTop": 2,
                                  "textAlign": "center",
                                }
                              }
                            >
                              Create an account
                            </Text>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenContainer>
  </RNCSafeAreaProvider>
</View>
`;
